### ISSUE TITLE
Don’t run CI if only .my files are pushed

### ISSUE THREAD

**Issue by ann0see** [jamulussoftware/jamulus#1806]
**Has this feature been discussed and generally agreed?**

No 

**Describe the solution you'd like**

The CI should not run if only .md files are changed since this is just a waste of ressources.
OBS seems to do it: https://github.com/obsproject/obs-studio/blob/master/.github/workflows/main.yml
**Describe alternatives that have been considered**

**Comment by pljones** [comment]
Surely things like the deb src file need the updated `.md` files as well as all the other source?  That means running the Debian build, at least.

**Comment by ann0see** [comment]
Not sure I understand. The deb src deb is not created via GH actions, as far as I know?

**Comment by pljones** [comment]
If we're ever to become the official source of the deb files, rather than relying on them being repacked, then we need to.

**Comment by hoffie** [comment]
> If we're ever to become the official source of the deb files, rather than relying on them being repacked, then we need to.

But would this require that each change of a Github-related file (.github/) or an in-repo documentation (SECURITY.md) must trigger code checks and build artifacts? That is what this PR would stop from working (and therefore make those PRs quicker).

Build artifacts (e.g. when triggered by code changes or tagging) would still include those files, as far as I understand.

**Comment by pljones** [comment]
The Clang format check runs in under 20 seconds and is run once for the change.  The builds take "ages".  The CodeQL checks don't take up much of that, from what I'm aware.  However, it's not that easy to see, as it's run in the same workflow.

**Comment by ann0see** [comment]
> The CodeQL checks don't take up much of that, from what I'm aware. 

Yes, they do. If you disable codeql the build is speeded up by a factor of 2 as far as I remember

**Comment by ann0see** [comment]
> It would be better to split out the Clang format and CodeQL checks into separate workflows triggered only by code changes...

They’re already in different files.

**Comment by pljones** [comment]
> They’re already in different files.

The Clang format check is a separate run.  It runs once, not six times.

The CodeQL runs for each build.  It _shouldn't_ take as long as it does.  It takes that long because it duplicates a lot of what else happens, IIRC.

**Comment by ann0see** [comment]
Actually, I think it's even a bit more complicated. On macOS we're doing 3 builds:

1. CodeQl
2. Packaging
3. Legacy packaging

Clang format does not yet abort packaging/CodeQl

We are - as far as I remember - running CodeQl while packaging is done. So except for macOS we're close to the suggested workflow. If that's not the case, please show where we can avoid duplication (except for macOS, ofc.)

**Comment by pljones** [comment]
> Clang format does not yet abort packaging/CodeQl

It doesn't abort at all, as it's a separate work flow.  (That is, it doesn't stop any other work flow running - it breaks the build, of course, if it fails.)

> We are - as far as I remember - running CodeQl while packaging is done.

As far as I can see, CodeQL happens after upload of the final artifacts.  Which is rather late in the case it finds a problem, of course.

**Comment by ann0see** [comment]
> As far as I can see, CodeQL happens after upload of the final artifacts. 

https://github.com/jamulussoftware/jamulus/blob/27f9a35b141829c3c5966f36af4df9a6705c5559/.github/workflows/autobuild.yml#L169-L178

Isn't CodeQl monitoring the build process? Tagging @nefarius2001 here for clarification.

> It doesn't abort at all, as it's a separate work flow. (That is, it doesn't stop any other work flow running (being autobuild.yml) - it breaks the build, of course, if it fails.)

Correct. However, we could stop building if clang-format fails.

**Comment by pljones** [comment]
The "Initialize" step doesn't do much other than install the tool, as far as I know.  That should run once.  Then the separate builds should have run the analyse step before moving on to publish.
